lib/dirList.js

```diff
@@ -10,36 +11,82 @@ const dirList = {
    * @param {function(error, entries)} callback
    * note: can't use glob because don't get error on non existing dir
    */
-  list: function (dir, callback) {
+  list: async function (dir, options) {
     const entries = { dirs: [], files: [] }
-    fs.readdir(dir, (err, files) => {
-      if (err) {
-        return callback(err)
-      }
-      if (files.length < 1) {
-        callback(null, entries)
+    const files = await fs.readdir(dir)
+    if (files.length < 1) {
+      return entries
+    }
+
+    await Promise.all(files.map(async filename => {
```

Comment:
please use p-limit here. We are not really interested in the result of all the promises.

Reply:
done